vm/src/obj/objdict.rs

```diff
@@ -247,6 +234,54 @@ impl ItemProtocol for PyDictRef {
     }
 }
 
+// Implement IntoIterator so that we can easily iterate dictionaries from rust code.
+impl IntoIterator for PyDictRef {
```

Comment:
Rather than implementing this for the specific type, could this be implemented for `&PyDict`, and then a blanket impl for `PyRef<T>` (i.e. something like `impl<T> IntoIterator for PyRef<T> where &T: IntoIterator`).

Reply:
Good tip. I think we want the other way around: ie. ```impl<T> IntoIterator for &PyRef<T>```, otherwise you always clone.

Reply:
Or maybe, not... you mean PyDict not PyDictRef.

Reply:
But, I don't think you implement it for ```&PyDict``` because you need the PyRef in order to clone.

Reply:
Why do you need to clone? Why not:

```rust
struct DictIter<'a> {
    dict: &'a PyDict,
    ...
}
```

Reply:
I'll give that a go. Thanks.

Reply:
I get as far as this

```
impl<T> IntoIterator for PyRef<T>
where
    T: IntoIterator + PyValue,
{
    type Item = <T as IntoIterator>::Item;
    type IntoIter = <T as IntoIterator>::IntoIter;

    fn into_iter(self) -> Self::IntoIter {
        self.deref().into_iter()
    }
}
```

which gives:

```
266 |         self.deref().into_iter()
    |         ^^^^^^^^^^^^ cannot move out of borrowed content
```

Reply:
I think I'll merge what I've got here and open a separate PR for this improvement.